### PR TITLE
Simple parquet-tools bin

### PR DESCRIPTION
**Pull request by elferherrera**
Parquet tool companion for the parquet2 crate. This bin creates an executable that can be used to extract information from a parquet file. It has the next subcommands:

- rowcount: show row count from parquet file
- meta: shows metadata from file
- dump: sample data from columns

### CODE REVIEW

**Comment by elferherrera** [comment]
@jorgecarleitao I was checking that the bin is failing codecov tests. However, since it is a binary I didnt add test functions. What do you suggest?

File: parquet-tools/src/dump.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+// Dumps data from the file.
+// The function prints a sample of the data from each of the RowGroups.
+// The the columns to be printed and the sample size is controlled using the
+// arguments introduced in the command line
+pub fn dump_file<T>(file_name: T, matches: &ArgMatches) -> Result<()>
+where
+    T: AsRef<Path> + std::fmt::Display,
+{
+    let mut file = File::open(file_name).unwrap();
```
Comment (jorgecarleitao) [inline parquet-tools/src/dump.rs:22]: unwrap?

**Review by jorgecarleitao — COMMENTED** [review]
Sorry it took me so long. This looks great. 👍

I think that we could benefit from unit-tests. One way this is usually done is to declare a `lib.rs` with tests, that contains the rust API, and then have the CLI be the binary that calls that API.

E.g. have the API receive a `fmt` that formats itself (a bit like when `fmt::Debug` is implemented). This way we can test that the output of the CLI is correct, and allow us to remove the `println!` statements that are difficult to test.

**Comment by elferherrera** [comment]
@jorgecarleitao Hi, I did a restructure of the crate to have the parquet functions as library functions with tests. Now the functions take a writer to generate the reports. This means that in the future, if we would like to save the reports to files we could pass a file writer and that would work better. 

Let me know what you think

**Comment by jorgecarleitao** [comment]
Thanks a lot, looks great! Good to go 👍

3 questions:

1. I think we need to have the CI in place to test this, right?
2. I can't merge due to conflicts; I think that it is due to a merge commit on this PR. Could you rebase?
3. I may move this to Apache Arrow at some point. Since this is a significant change, would you be open to sign a [`ICLA`](https://www.apache.org/licenses/contributor-agreements.html#clas), so that in the event of a donation, the process is a bit easier?

**Comment by elferherrera** [comment]
> 1. I think we need to have the CI in place to test this, right?
It could be a good idea to run the tests in the crate to see if there a changes in the expected format for the reports.

> 2. I can't merge due to conflicts; I think that it is due to a merge commit on this PR. Could you rebase?
Done 🚀

> 3. I may move this to Apache Arrow at some point. Since this is a significant change, would you be open to sign a [`ICLA`](https://www.apache.org/licenses/contributor-agreements.html#clas), so that in the event of a donation, the process is a bit easier?
Sure, no problem

**Comment by jorgecarleitao** [comment]
Thanks a lot, @elferherrera ; this is now ready to merge 🎉 ; could you rebase?

**Comment by elferherrera** [comment]
@jorgecarleitao I've just saw your message. The branch has been rebased

**Comment by jorgecarleitao** [comment]
I can't merge yet. Could you rebase and squash, so that it has a single non-merge commit?

**Comment by elferherrera** [comment]
> I can't merge yet. Could you rebase and squash, so that it has a single non-merge commit?

Done. I have squashed all commits

**Comment by jorgecarleitao** [comment]
Merged. 🎉  Thanks a lot, @elferherrera !

**Pull request merged**